guillotina/json/deserialize_content.py

```diff
@@ -117,11 +117,14 @@ async def set_schema(
             else:
                 data_value = data[name] if name in data else None
                 found = True if name in data else False
-            if found:
 
-                if not self.check_permission(write_permissions.get(name)):
+            if found or field.missing_value is not None:
+                if found and not self.check_permission(write_permissions.get(name)):
                     raise Unauthorized("Write permission not allowed")
 
+                if not found and field.missing_value is not None:
+                    data_value = deepcopy(field.missing_value)
+
```

Comment:
```suggestion
            # Only set missing_value when it's not present in 'data' and we are validating_all fields
            must_set_mv = not found and validate_all and field.missing_value is not None

            if found or must_set_mv:
                if found and not self.check_permission(write_permissions.get(name)):
                    raise Unauthorized("Write permission not allowed")

                if must_set_mv:
                    data_value = deepcopy(field.missing_value)

```

Reply:
Its better you do a commit than yourself a suggestion its difficult to comment on specific line. This change may involve a missing value on a field that user does not have permission to set which will return on a 401.

Reply:
already commited ;-)

>  This change may involve a missing value on a field that user does not have permission to set which will return on a 401.

We only check the permission if the field is found in the data.